addons/website_crm_partner_assign/static/src/js/crm_partner_assign.js

```diff
@@ -1,202 +1,235 @@
 'use strict';
 
-var rpc = require('web.rpc');
-var weContext = require('web_editor.context');
-require('web.dom_ready');
+var sAnimations = require('website.content.snippets.animation');
+
 /*
  * This file is intended to add interactivity to survey forms rendered by
  * the website engine.
  */
 
-var interested_form = $('.interested_partner_assign_form');
-var desinterested_form = $('.desinterested_partner_assign_form');
-var opp_stage_buttons = $('.opp-stage-button');
-var new_opp_form = $('.new_opp_form');
+sAnimations.registry.crmPartnerAssign = sAnimations.Class.extend({
+    selector: "#portal_my_lead:has(.interested_partner_assign_form,.desinterested_partner_assign_form,.opp-stage-button,.new_opp_form')",
+    read_events: {
+        'click .interested_partner_assign_confirm': '_onInterestedPartnerAssignConfirm',
+        'click .desinterested_partner_assign_confirm': '_onDesinterestedPartnerAssignConfirm',
+        'click .opp-stage-button': '_onOppStageButtons',
+        'change .edit_contact_form .country_id': '_onEditContactForm',
+        'click .edit_contact_confirm': '_onEditContactConfirm',
+        'click .new_opp_confirm': '_onNewOppConfirm',
+        'click .edit_opp_confirm': '_onEditOppConfirm',
+        'change .edit_opp_form .next_activity': '_onEditOppConfirm',
+        'click div.input-group span.fa-calendar': '_onCalendarIcon'
+    },
 
-if(!interested_form.length && !desinterested_form.length && !opp_stage_buttons.length && !new_opp_form.length) {
-    return $.Deferred().reject("DOM doesn't contain website_crm_partner_assign elements");
-}
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
 
-$('.interested_partner_assign_confirm').on('click',function(e){
-    var $btn = $(this);
-    if ($('.interested_partner_assign_form .comment_interested').val() && $('.interested_partner_assign_form .contacted_interested').prop('checked')){
+    /**
+     * @override
+     * @param {Object} ev
+     */
+    _onInterestedPartnerAssignConfirm: function (ev) {
+        var $btn = $(ev.currentTarget);
+        if ($('.interested_partner_assign_form .comment_interested').val() && $('.interested_partner_assign_form .contacted_interested').prop('checked')){
+            $btn.prop('disabled', true);
+            this._rpc({
+                    model: 'crm.lead',
+                    method: "partner_interested",
+                    args: [
+                        [parseInt($('.interested_partner_assign_form .assign_lead_id').val())],
+                        $('.interested_partner_assign_form .comment_interested').val()
+                    ],
+                })
+                .then(function (){
+                    window.location.href = '/my/leads';
+                })
+                .always(function () {
+                    $btn.prop('disabled', false);
+                });
```

Comment:
I know you kept as it was before but this is not the way we indent by convention. This should be like this:
```javascript
this._rpc({
    // ...
}).then(function () {
    // ...
}).always(function () {
    // ...
});
```